Automattic/WP-Job-Manager
# Automate Release (Core)

### Changes proposed in this Pull Request

* Add an `npm run release <version>` command
* Adapt Sensei's release script to prepare a new release, bump versions, replace placeholders
* Create provisional changelog based on new 'Release Notes' section in every PR for the version's milestone
* Create release PR
* Add an action triggered by merging a PR from a `release/*` branch, that writes the final changelog entry, creates and tags a new release on GH, then deploys it to WordPress.org

### Testing instructions

* See testing repo:
  * Generated release PR: https://github.com/yscik/WP-Job-Manager/pull/7
  * Release: https://github.com/yscik/WP-Job-Manager/releases
  * Action log: https://github.com/yscik/WP-Job-Manager/actions/runs/6152058881/job/16693423980?pr=7
  * (Wordpress.org deployment is set up as a dry run for testing )
* We can test further with the upcoming release

### Usage docs

* [GitHub CLI ](https://cli.github.com/) is required for the initial release command. Use `gh auth login` the first time to set it up
* Release will be done from the current branch, which usually should be an up-to-date trunk
* A milestone needs to be set up for the version being released, and merged non-maintenance PRs should be included in it
